src/main/java/frc/robot/commands/ArmCommand.java

```diff
@@ -115,7 +121,13 @@ public void execute() {
         topReached = true;
       }
       if (topReached) {
-        ArmSubsystem.getInstance().controlTopArmBrake(true);
+        // debouncer to increase the time for arm to reach set point (only if its in pickup position)
+        if (armSetpoint == ArmSetpoint.PICKUP) {
+          ArmSubsystem.getInstance().controlTopArmBrake(m_BrakeDebounce.calculate(topReached));
```

Comment:
I just thought of an edge case here.
The conditions for starting the backwards voltage is (topReached && botReached)
The conditions for triggering the top brake is just topReached.

This means if the top arm reaches the setpoint before the bottom arm, it could trigger the brake well before the backwards voltage is applied.

All you need to do is change line 126 to this:
```
if (armSetpoint == ArmSetpoint.PICKUP) {
  if (bottomReached) {
    ArmSubsystem.getInstance().controlTopArmBrake(m_BrakeDebounce.calculate(topReached));
  } else {
etc..
```

This way the top brake in PICKUP will only be applied when both arms have reached their setpoint and the debounce has happened. However for other setpoints, the top brake can still trigger earlier than the bottom brake if it has reached it setpoint ahead of the bottom.